celery/app/routes.py

```diff
@@ -38,14 +38,20 @@ def __init__(self, map):
             else:
                 self.map[k] = v
 
+        # We sort by the regex pattern's length since longer regex patterns
+        # are likely to be more specific.
+        self.patterns = tuple(reversed(sorted(tuple(self.patterns.items()),
```

Comment:
Why the explicit cast from `.items()` to `tuple`?

```python
>>> sorted({1:1,3:3,2:2}.items())
[(1, 1), (2, 2), (3, 3)]
```

Reply:
Because sorted expects it. Maybe iter() would also be sufficient.

Reply:
It didn't. I was wrong and you're right.
Thanks.